packages/notebook/src/widget.ts

```diff
@@ -1367,6 +1371,44 @@ export class Notebook extends StaticNotebook {
     this.mode = 'edit';
   }
 
+  private _onJumpRequest(
+    editor: CodeEditor.IEditor,
+    jump: CodeEditor.IJump
+  ): void {
+    // Using `index = this._findCell(editor.host)` does not work,
+    // as the host editor has not switched to the clicked cell yet.
+    let index;
+
+    // The mouse event is utilized to workaround Firefox's issue.
+    if (jump.mouseEvent !== undefined) {
+      index = this._findTargetCell(jump.mouseEvent).index;
+    } else {
+      index = this._findCell(jump.origin);
+    }
+
+    let { token, cellIndex } = this._findLastDefinition(jump.token, index);
+
+    // nothing found
+    if (!token) {
+      return;
+    }
+
+    // Prevents event propagation issues
+    setTimeout(() => {
```

Comment:
What is the purpose for doing this asynchronously? I don't understand the comment above.